Set smoothing in realignment as variable

dagra opened the pull request:
Hi,
I wanted the option to use a different smoothing function during realignment, so i added an argument in the corresponding class (and function) that represents the desired smoothing function to apply. I think this can be useful to others too, hence the PR.

The default value is set to pypreprocess' smoothing, so the current functionality doesn't change.

As an example, in order to use nibabel's smoothing:
```
from nibabel.processing import smooth_image

mrimc = MRIMotionCorrection(smooth_func=smooth_image)
```

bthirion commented on pypreprocess/realign.py:
```diff
@@ -304,7 +304,8 @@ class MRIMotionCorrection(object):
     def __init__(self, sep=4, interp=3, fwhm=5., quality=.9, tol=1e-8,
-                 lkp=None, verbose=1, n_iterations=64, n_sessions=1):
+                 lkp=None, verbose=1, n_iterations=64, n_sessions=1,
+                 smooth_func=smooth_image):
```
Please update (or create ;-) ) the dosctring accordingly.
Thx

bthirion commented:
This does not seem to harm, but  I don't really understand the use case. In terms of efficiency, it should be way better to implement the standard smoothing ?

dagra commented:
> This does not seem to harm, but I don't really understand the use case. In terms of efficiency, it should be way better to implement the standard smoothing ?

In my experiments, using nibabel's smoothing, the estimation is almost 17% faster (e.g. from 270 seconds to 225 seconds), which is significant in my case.

bthirion commented:
OK. Can you also add a test ?

dagra commented:
> OK. Can you also add a test ?

I added a test, but i wasn't sure what kind of test you'd want. I decided to test it as you already test the motion correction fit, so i used the corresponding test. Also, i wasn't sure if you didn't want any changes, so i copied the code and just replaced with nibabel's smoothing, but there is now some duplicate code which can be factored out. If you don't like the duplicate code, i can refactor it.

bthirion commented:
Thx for adding the test. A refactoring would be welcome if you have time to keep the text well-structured.
Any other review ? Otherwise, I'll merge it this week.

dagra commented:
I'm sorry for the delayed response. I think the code is clean now.

dimrozakis commented:
Hey @bthirion 

Could you please go ahead and merge this PR? Thanks!
